etp-front/src/pages/valvonta-kaytto/toimenpiteet.js

```diff
@@ -483,3 +483,135 @@ export const osapuoliHasHallintoOikeus = (toimenpide, osapuoli) =>
+        ];
+        break;
+
+      case type['decision-order']['notice-first-mailing']:
+        allowedToimenpidetypes = [
+          type['decision-order']['notice-first-mailing'],
+          type['decision-order']['notice-second-mailing']
+        ];
+        break;
+
+      case type['decision-order']['notice-second-mailing']:
+        allowedToimenpidetypes = [
+          type['decision-order']['notice-second-mailing'],
+          type['decision-order']['notice-bailiff'],
+          type['decision-order']['waiting-for-deadline']
+        ];
+        break;
+
+      case type['decision-order']['notice-bailiff']:
+        allowedToimenpidetypes = [
+          type['decision-order']['waiting-for-deadline']
+        ];
+        break;
+
+      case type['decision-order']['waiting-for-deadline']:
+        allowedToimenpidetypes = [
+          type['court-hearing'],
+          type['penalty-decision']['hearing-letter']
+        ];
+        break;
+
+      case type['court-hearing']:
+        allowedToimenpidetypes = [
+          type['penalty-decision']['hearing-letter'],
+          type['penalty-list-delivery-in-progress']
+        ];
+        break;
+
+      case type['penalty-decision']['hearing-letter']:
+        allowedToimenpidetypes = [
+          type['penalty-decision']['hearing-letter'],
+          type['penalty-decision']['actual-decision']
+        ];
+        break;
+
+      case type['penalty-decision']['actual-decision']:
+        allowedToimenpidetypes = [
+          type['penalty-decision']['actual-decision'],
+          type['penalty-decision']['notice-first-mailing']
+        ];
+        break;
+
+      case type['penalty-decision']['notice-first-mailing']:
+        allowedToimenpidetypes = [
+          type['penalty-decision']['notice-first-mailing'],
+          type['penalty-decision']['notice-second-mailing'],
+          type['penalty-decision']['waiting-for-deadline']
+        ];
+        break;
+
+      case type['penalty-decision']['notice-second-mailing']:
+        allowedToimenpidetypes = [
+          type['penalty-decision']['notice-second-mailing'],
+          type['penalty-decision']['notice-bailiff'],
+          type['penalty-decision']['waiting-for-deadline']
+        ];
+        break;
+
+      case type['penalty-decision']['notice-bailiff']:
+        allowedToimenpidetypes = [
+          type['penalty-decision']['waiting-for-deadline']
+        ];
+        break;
+
+      case type['penalty-decision']['waiting-for-deadline']:
+        allowedToimenpidetypes = [
+          type['court-hearing'],
+          type['penalty-list-delivery-in-progress']
+        ];
+        break;
+
+      case type['penalty-list-delivery-in-progress']:
+        allowedToimenpidetypes = [type.closed];
+    }
+
+    // Valvonnan lopetus is always allowed
+    allowedToimenpidetypes.push(type.closed);
+
+    return R.filter(
+      R.compose(R.includes(R.__, allowedToimenpidetypes), R.prop('id')),
+      toimenpidetypes
+    );
+  }
+);
```

Comment:
Tässähän kai voisi käyttää Ramdan [cond](https://ramdajs.com/docs/#cond). Saattaisi olla parempi niin, mutta varmaan ihan hyvä näinkin.

Reply:
Se ei mun mielestä oikein tuonyt tähän tapaukseen mitään arvoa, koska kuten usein ramdan kanssa, siinä sekä ehdot että paluuarvot on kääritty funktioihin ja tässä switch case hoitaa suoraan sen vertailun ja casessa tehtävä operaatio on vain arvon asettaminen, joten mielestäni tämä meni kaikista yksinkertaisimmin näin JS:n natiiviominaisuuksilla. Muihin tapauksiin ehdottoman käyttökelpoinen kyllä.